ios/RNViewShot.m

```diff
@@ -47,12 +47,12 @@ - (dispatch_queue_t)methodQueue
 
     // Get view
     UIView *view;
-
-    if ((int)target == -1) {
+    view = viewRegistry[target];
+    BOOL nativeCapture = false;
+    if (!view) {
```

Comment:
mmh I don't think this should be a fallback of not having `viewRegistry[target]`. Potentially target is wrong and no longer in the registry and we still want to detect that potential wrong target value and trigger an error (instead of fallbacking to snapshotting the screen, ultimately that could end up leaking user's screen in production!) (so I think it's better to keep the following `if(!view)` check code that throw in such case)

so I think doing `[target intValue] == -1` would be better.

Reply:
That's a very good point, I've made the necessary changes to keep the thrown error in place.